packages/wonder-blocks-layout/components/media-layout.js

```diff
@@ -0,0 +1,242 @@
+        for (const styleSize of Object.keys(styleSheets)) {
+            const styleSheet = styleSheets[styleSize];
+
+            if (!styleSheet) {
+                continue;
+            }
+
+            // And then through each key of each stylesheet
+            for (const name of Object.keys(styleSheet)) {
+                if (mockStyleSheet.hasOwnProperty(name)) {
+                    continue;
+                }
+
+                // We create an entry that combines the values from all of
+                // the stylesheets together in least-specific to most-specific
+                // priority (thus small/medium/large styles will always have
+                // precedence over "all" or mdOrSmaller/mdOrLarger/etc.).
+                mockStyleSheet[name] = [
+                    styleSheets.all && styleSheets.all[name],
+                    mediaSize === "small" && [
+                        styleSheets.mdOrSmaller &&
+                            styleSheets.mdOrSmaller[name],
+                        styleSheets.small && styleSheets.small[name],
+                    ],
+                    mediaSize === "medium" && [
+                        styleSheets.mdOrSmaller &&
+                            styleSheets.mdOrSmaller[name],
+                        styleSheets.mdOrLarger && styleSheets.mdOrLarger[name],
+                        styleSheets.medium && styleSheets.medium[name],
+                    ],
+                    mediaSize === "large" && [
+                        styleSheets.mdOrLarger && styleSheets.mdOrLarger[name],
+                        styleSheets.large && styleSheets.large[name],
+                    ],
+                ];
+            }
+        }
+
+        return mockStyleSheet;
+    }
+
+    render() {
+        const {children} = this.props;
+
+        // We listen to the MediaLayoutContext to see what defaults we're
+        // being given (this can be overriden by wrapping this component in
+        // a MediaLayoutContext.Consumer).
+        return (
+            <MediaLayoutContext.Consumer>
+                {({overrideSize, ssrSize, mediaSpec}) => {
+                    // We need to figure out what the current media size is
+                    // If an override has been specified, we use that.
+                    // If we're rendering on the server then we use the default
+                    // SSR rendering size.
+                    // Otherwise we attempt to get the current size based on
+                    // the current MediaSpec.
+                    const mediaSize =
+                        overrideSize ||
+                        (this.isServerSide() && ssrSize) ||
+                        this.getCurrentSize(mediaSpec);
+
+                    // Generate a mock stylesheet
+                    const styles = this.getMockStyleSheet(mediaSize);
+
+                    return children({mediaSize, mediaSpec, styles});
+                }}
```

Comment:
It's only a small method, but I think moving this to a separate method rather than inline would be cleaner. Thoughts?